tests/_test-utils/mockeditor.js

```diff
@@ -0,0 +1,66 @@
+/* global setTimeout */
+
+const testEditor = {
+	on: addListener(),
+
+	once: addListener( true ),
+
+	fire( evtName, evt ) {
+		const listeners = this._listeners[ evtName ];
+
+		if ( !listeners ) {
+			return;
+		}
+
+		this._listeners[ evtName ] = listeners.filter( ( { callback, once } ) => {
+			callback( evt );
+			return !once;
+		} );
+	},
+
+	setData( data, { callback } = {} ) {
+		callback && callback();
+	},
+
+	getData() {
+		return '';
+	},
+
+	setReadOnly() {
```

Comment:
:thinking:

Reply:
Usually you write more verbose comments 😜

Reply:
Sorry, I thought it's some leftover as this function is empty. Somehow missed filename :smile:

Reply:
At first it made writing tests a lot easier, especially before `t/3115`. Initially I wasn't sure if mocked editor is only for start, or for longer. But now when integration is nicely shaped, not much changes are expected, so I can replace mock editor with real CKEditor.